nautobot/extras/models/customfields.py

```diff
@@ -251,6 +309,12 @@ class CustomField(BaseModel, ChangeLoggedModel, NotesMixin):
         limit_choices_to=FeatureQuery("custom_fields"),
         help_text="The object(s) to which this field applies.",
     )
+    grouping = models.CharField(
+        max_length=50,
```

Comment:
50 is pretty limiting. JOB_MAX_GROUPING_LENGTH is 255, which may be more than needed, but storage isn't that expensive.